### ISSUE TITLE
Add a "setup option/warning" to download rootfs image

### ISSUE THREAD

**Issue by pintaf** [maruos/vendor_maruos_packages_apps_MaruSettings#2]
In order to reduce system image size, it would be nice if the rootfs image could be downloaded after boot.

Hence an option must be added to do this in the settings, and a next step would be to choose it's rootfs flavor. Debian, Ubuntu...
After downloading, it should also be extracted thx to mcprepare or similar

**Comment by utzcoz** [comment]
@pintaf @pdsouza I think this feature can be started.

**Comment by pdsouza** [comment]
@utzcoz Yes, we need this feature to build for our older devices with smaller system partitions. The Nexus 5 for example will not build right now even with the minimal desktop image because Android 9 takes up too much space on `/system`.

**Comment by utzcoz** [comment]
Okay, let's kick off it. I will think potential solutions, and post it here for discussion if got one. If you have solution, we can discuss it firstly.

**Comment by pdsouza** [comment]
For hosting the rootfs image, we can make use of GitHub Releases like how maru-0.6 had releases hosted. As part of the automated builds, I make sure to upload just the desktop rootfs image along with the device images, for example the arm64 image for our v0.6.8 is maru-geb4a1b8-debian-stretch-arm64-rootfs-e990e414.tar.gz and is listed here: https://github.com/maruos/builds/releases/tag/v0.6.8.

So the download logic of the app can just check for the version matching the current build version and architecture and download the right image by just checking the filename of the release assets for that build version.

After that we just need to move it to `/data/maru/containers` and unpack it correctly using `mcprepare`. Is this possible with system-privileged apps?

**Comment by pintaf** [comment]
Sorry guys, I would love to participate in this, but I see no time do to so before at least 2 months. Though times at work... 

Le 2 juin 2021 18:57:18 GMT+02:00, pdsouza ***@***.***> a écrit :
>For hosting the rootfs image, we can make use of GitHub Releases like
>how maru-0.6 had releases hosted. As part of the automated builds, I
>make sure to upload just the desktop rootfs image along with the device
>images, for example the arm64 image for our v0.6.8 is
>maru-geb4a1b8-debian-stretch-arm64-rootfs-e990e414.tar.gz and is listed
>here: https://github.com/maruos/builds/releases/tag/v0.6.8.
>
>So the download logic of the app can just check for the version
>matching the current build version and architecture and download the
>right image by just checking the filename of the desktop image.
>
>After that we just need to move it to `/data/maru/containers` and
>unpack it correctly using `mcprepare`. Is this possible with
>system-privileged apps?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/maruos/vendor_maruos_packages_apps_MaruSettings/issues/2#issuecomment-853215649

-- Envoyé de /e/ Mail.

**Comment by pdsouza** [comment]
> Sorry guys, I would love to participate in this, but I see no time do to so before at least 2 months. Though times at work...

No worries @pintaf, work always tends to get in the way when you least expect it! But looking forward to having you helping out with Maru again when you are able to - I think it will be fun for you to be involved in the 0.7 release after all the early work you did for it!

**Comment by pintaf** [comment]
Yes I would have loved to work on the UI permitting to dl the desktop images as I have a past as android dev, but we'll see, maybe I'll find some time. But I always get an eye on this and truly believe maru is the future. 

Le 2 juin 2021 20:27:25 GMT+02:00, pdsouza ***@***.***> a écrit :
>> Sorry guys, I would love to participate in this, but I see no time do
>to so before at least 2 months. Though times at work...
>
>No worries @pintaf, work always tends to get in the way when you least
>expect it! But looking forward to having you helping out with Maru
>again when you are able to - I think it will be fun for you to be
>involved in the 0.7 release after all the early work you did for it!
>
>
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/maruos/vendor_maruos_packages_apps_MaruSettings/issues/2#issuecomment-853282416

-- Envoyé de /e/ Mail.
